File: Python/37.py
# Python function to compute GCD of two numbers using recursion
def gcd(a,b):
    if(b==0): # Base case
        return a
    else:
        return gcd(b,a%b) # Recursive call

# Tester code
a=int(input("Enter first number:"))
b=int(input("Enter second number:"))
GCD=gcd(a,b)
print("GCD is: ")
print(GCD)

File: Java/37.java
  // To find G.C.D using Recursion
  
  import java.util.*;
  class Sol
{
    // Recursive function 
    static int gcd(int a, int b)
    {
      if (b == 0)
        return a;
      return gcd(b, a % b); 
    }
    public static void main(String args[]){
      Scanner sc=new Scanner(System.in);
      int a=sc.nextInt();
      int b=sc.nextInt();
      System.out.println(gcd(a,b));
    }
}
